dotcom-rendering/src/web/components/Article.tsx

```diff
@@ -44,6 +49,14 @@ export const Article = ({ CAPIArticle, NAV, format }: Props) => {
 						background: ${brandAlt[400]};
 						color: ${neutral[7]};
 					}
+
+					/* We apply this style when the side navigation is open the prevent the document body from scrolling */
+					/* See Nav.tsx */
+					.nav-is-open {
```

Comment:
Bit of a shame that we need to use a CSS class for this, I'm not entirely sure if theres another solution for this though. Anyone got any thoughts?

Reply:
This does couple `Article.tsx` to it's child `Nav.tsx`, which is a shame.

I can think of a few options:
- rename `nav-is-open` to a more utility style name like `no-scroll`, `lock-scroll` or even `overflow-hidden`. While this seems like a superficial change, it moves the responsibility of this functionality, locking the scrolling on the body, to the `Article.tsx`, exposing it to be used by other components.
- remove the coupling completely and keep everything in `Nav.tsx`. We could achieve this by adding a `style` tag in `Nav.tsx` that does all the `nav-is-open` business, keeping it in one place. This is also more explicit than the above option as making it more abstract isn't great, as we don't actually have more use cases for it.

Reply:
Actually its coupling `Article.tsx`, `Nav.tsx`, and `VeggieBurgerMenu.tsx` 😰 

> rename nav-is-open to a more utility style name like no-scroll, lock-scroll or even overflow-hidden. While this seems like a superficial change, it moves the responsibility of this functionality, locking the scrolling on the body, to the Article.tsx, exposing it to be used by other components.

This would help quite a bit, but nav-is-open can't actually be all that generic... It needs to only apply `overflow: hidden` when the display is tablet or smaller, otherwise we'd be blocking scroll when the desktop navigation bar is open.

> remove the coupling completely and keep everything in Nav.tsx. We could achieve this by adding a style tag in Nav.tsx that does all the nav-is-open business, keeping it in one place. This is also more explicit than the above option as making it more abstract isn't great, as we don't actually have more use cases for it.

I quite like this idea, but it's still affected by the aforementioned issue of only blocking scrolling on a tablet sized device. If we were to apply the `overflow` style programatically we'd have to use Javascript to check what the screen size is before applying the style which comes with a host of its own issues.
Like the idea of making it more generic, it might make the CSS in VeggieBurgerMenu a bit confusing though, but can probably just resolve that with a comment.

What I could actually do is move the Global styles into `Nav.tsx`, that way atleast the styles are close to what uses them